x-pack/plugins/stack_connectors/server/connector_types/gen_ai/index.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import {
+  SubActionConnectorType,
+  ValidatorType,
+} from '@kbn/actions-plugin/server/sub_action_framework/types';
+import { GeneralConnectorFeatureId } from '@kbn/actions-plugin/common';
+import { urlAllowListValidator } from '@kbn/actions-plugin/server';
+import { ValidatorServices } from '@kbn/actions-plugin/server/types';
+import { assertURL } from '@kbn/actions-plugin/server/sub_action_framework/helpers/validators';
+import {
+  GEN_AI_CONNECTOR_ID,
+  GEN_AI_TITLE,
+  OpenAiProviderType,
+} from '../../../common/gen_ai/constants';
+import { GenAiConfigSchema, GenAiSecretsSchema } from '../../../common/gen_ai/schema';
+import { GenAiConfig, GenAiSecrets } from '../../../common/gen_ai/types';
+import { GenAiConnector } from './gen_ai';
+import { renderParameterTemplates } from './render';
+
+export const getConnectorType = (): SubActionConnectorType<GenAiConfig, GenAiSecrets> => ({
+  id: GEN_AI_CONNECTOR_ID,
+  name: GEN_AI_TITLE,
+  Service: GenAiConnector,
+  schema: {
+    config: GenAiConfigSchema,
+    secrets: GenAiSecretsSchema,
+  },
+  validators: [{ type: ValidatorType.CONFIG, validator: configValidator }],
+  supportedFeatureIds: [GeneralConnectorFeatureId],
+  minimumLicenseRequired: 'platinum' as const,
+  renderParameterTemplates,
+});
+
+export const configValidator = (
+  configObject: GenAiConfig,
+  validatorServices: ValidatorServices
+) => {
+  try {
+    assertURL(configObject.apiUrl);
+    urlAllowListValidator('apiUrl')(configObject, validatorServices);
+
+    if (
+      configObject.apiProvider !== OpenAiProviderType.OpenAi &&
+      configObject.apiProvider !== OpenAiProviderType.AzureAi
+    ) {
+      throw new Error('API Provider is not supported');
```

Comment:
Would probably be good to see the apiProvider value they passed in - that would show in the logs, well after a customer might have changed it later (using HTTP APIs to manage connectors/rules).